File: src/main/java/com/example/appjamback/global/security/principle/AuthDetailsService.java
package com.example.appjamback.global.security.principle;

import com.example.appjamback.domain.user.domain.User;
import com.example.appjamback.domain.user.domain.repository.UserRepository;
import com.example.appjamback.domain.user.exception.UserNotFoundException;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class AuthDetailsService implements UserDetailsService {

    private final UserRepository userRepository;

    @Override
    public UserDetails loadUserByUsername(String accountId) throws UsernameNotFoundException {
        User user = userRepository.findByAccountId(accountId)
                .orElseThrow(() -> UserNotFoundException.EXCEPTION);
        return new AuthDetails(user);
    }
}

File: src/main/kotlin/com/example/instargram/global/security/principle/AuthDetailsService.kt
package com.example.instargram.global.security.principle

import com.example.instargram.domain.user.domain.User
import com.example.instargram.domain.user.facade.UserFacade
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.stereotype.Service

@Service
class AuthDetailsService(
    private val userFacade: UserFacade
) : UserDetailsService {
    override fun loadUserByUsername(accountId : String?) : UserDetails {
        val user: User? = accountId?.let { userFacade.getUserByAccountId(it) }
        return AuthDetails(user!!.accountId)
    }
}
